File: project/.sdk/tm/py/utility/prepare_path.py
# ProjectName SDK utility: prepare_path

from __future__ import annotations
from utility.voxgig_struct import voxgig_struct as vs


def prepare_path_util(ctx):
    point = ctx.point

    parts = []
    if point is not None:
        p = vs.getprop(point, "parts")
        if isinstance(p, list):
            parts = p

    return vs.join(parts, "/", True)

File: project/.sdk/tm/lua/utility/prepare_path.lua
-- ProjectName SDK utility: prepare_path

local vs = require("utility.struct.struct")

local function prepare_path_util(ctx)
  local point = ctx.point

  local parts = {}
  if point ~= nil then
    local p = vs.getprop(point, "parts")
    if type(p) == "table" then
      parts = p
    end
  end

  return vs.join(parts, "/", true)
end

return prepare_path_util
